Annotate more fallthrough cases.

pkasting opened the pull request:
Bug: chromium:995993

dcommander commented:
Please address the following issues:
- The `/*FALLTHROUGH*/` comments should always start on Column 33.  That formatting convention is from libjpeg, which uses 8-space tabs to align comments vertically.  Those tabs were converted into spaces in libjpeg-turbo, but we retain the alignment.
- Please verify whether the fall-through annotations are necessary on cases that fall through to a `break` statement.  My testing suggests that they are not.  If you confirm, then please remove the annotations from those cases so as to avoid unnecessary code clutter.
- Please refer to the other Neon SIMD files for examples of how to correctly include **jconfigint.h**.  **jconfigint.h** is a generated file, so it will exist in the build directory, not the source directory.  Thus, referencing it using a path relative to the source directory is incorrect.  Also, it should be the first header included.

pkasting commented:
> * The `/*FALLTHROUGH*/` comments should always start on Column 33.  That formatting convention is from libjpeg, which uses 8-space tabs to align comments vertically.  Those tabs were converted into spaces in libjpeg-turbo, but we retain the alignment.

Ah, thanks for the explanation, I couldn't figure out the convention in the previous patch.  Fixed.

> * Please verify whether the fall-through annotations are necessary on cases that fall through to a `break` statement.  My testing suggests that they are not.  If you confirm, then please remove the annotations from those cases so as to avoid unnecessary code clutter.

I verified that these are necessary at least with our version of Clang.  "error: unannotated fall-through between switch labels" even on a "default:" that just contains a break;, if there is fall-through to it from code above.  So I have preserved these.

> * Please refer to the other Neon SIMD files for examples of how to correctly include **jconfigint.h**.  **jconfigint.h** is a generated file, so it will exist in the build directory, not the source directory.  Thus, referencing it using a path relative to the source directory is incorrect.  Also, it should be the first header included.

Changed to be pathless ("jconfigint.h") and first in all three files where I added it.
